syntax = "proto3";

package auth;

service AuthService {
    rpc Login(LoginRequest) returns (LoginResponse);
    rpc ValidateToken(TokenRequest) returns (ValidateResponse);
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message LoginResponse {
    bool success = 1;
    string token = 2;
    string message = 3;
}

message TokenRequest {
    string token = 1;
}

message ValidateResponse {
    bool valid = 1;
    string user_id = 2;
    string message = 3;
}